runtime/pkg/metricssql/time_range_parser.go

```diff
@@ -67,7 +67,43 @@ func (q *query) parseTimeRangeEnd(ctx context.Context, node *ast.FuncCallExpr) (
 	}, nil
 }
 
-func (q *query) parseTimeRangeArgs(args []ast.ExprNode) (duration.Duration, int, string, error) {
+func (q *query) getWatermark(ctx context.Context, colName string) (watermark time.Time, err error) {
```

Comment:
Would it be possible to leverage the `Watermark` function from the `metricsview` package? 

https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/metricsview/executor.go#L82-L86

Reply:
Actually in the current time range functions we also allow user to provide an optional column whereas in the metricview only time dimension can be used. 
We can restrict the time range functions as well to work only with `timedimension` and then we can leverage the pkg function itself.